x/market/client/cli/flags.go

```diff
@@ -0,0 +1,45 @@
+package cli
+
+import (
+	"bytes"
+	"encoding/json"
+	flag "github.com/spf13/pflag"
+)
+
+const (
+	FlagDealFile = "deal-json-file"
+)
+
+type createDealInputs struct {
+	DataSchema            []string `json:"data_schema"`
+	Budget                string   `json:"budget"`
+	WantDataCount         uint64   `json:"want_data_count"`
+	TrustedDataValidators []string `json:"trusted_data_validators"`
+}
+
+type XCreateDealInputs createDealInputs
+
+type XCreateDealInputsExceptions struct {
+	XCreateDealInputs
+	Other *string
+}
+
+func (input *createDealInputs) UnmarshalJSON(data []byte) error {
+	var createDealEX XCreateDealInputsExceptions
+	decoder := json.NewDecoder(bytes.NewReader(data))
+	decoder.DisallowUnknownFields()
+
+	if err := decoder.Decode(&createDealEX); err != nil {
+		return nil
+	}
+
+	*input = createDealInputs(createDealEX.XCreateDealInputs)
+	return nil
+}
+
+func FlagSetCreateDeal() *flag.FlagSet {
+	fs := flag.NewFlagSet("", flag.ContinueOnError)
```

Comment:
What does this settings mean?
I wonder what changes without this setting.

Reply:
I guess this is the general pattern to create an unnamed flag set using https://github.com/spf13/pflag (if we don't need any name for the flag set).

Reply:
Anyway, in my opinion, we don't need to follow the Osmosis' way. 

For consistency, I'd suggest to define flags one by one using: `cmd.Flags().String(...)`: https://github.com/medibloc/panacea-core/blob/388585acad2058b5de0c98ef615f8ed41eb286b6/x/aol/client/cli/txTopic.go#L42

When upgrading cosmos-sdk to v0.44, we can change all modules to use `FlagSet`.

Reply:
Oh, This create a flagset without any name. 
I understand. 😀

Reply:
I agree with that we need to follow Osmosis' way and not yet to use ```FlagSet```. Thanks for your feedback.